File: server/BarAPP/src/main/java/org/pbarreiro/barapp/dto/ProductoDTO.java
package org.pbarreiro.barapp.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductoDTO {
    private Long id;

    @NotBlank(message = "El nombre es obligatorio")
    private String nombre;

    @NotNull(message = "El precio es obligatorio")
    @DecimalMin(value = "0.0", inclusive = false, message = "El precio debe ser mayor que cero")
    private BigDecimal precio;

    private Integer stock;
    private Long categoriaId;
    private CategoriaDTO categoria;
    private String imagenUrl;
    private Boolean activo;
}

File: mobile_client/BarAPP/composeApp/src/commonMain/kotlin/org/pabarreiro/barapp/data/remote/dto/ProductoDTO.kt
package org.pabarreiro.barapp.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class ProductoDTO(
    val id: Long? = null,
    val nombre: String,
    val precio: Double,
    val stock: Int? = null,
    val categoriaId: Long? = null,
    val imagenUrl: String? = null,
    val activo: Boolean? = true
)
